uk.co.compucorp.civicrm.hrcomments/CRM/HRComments/BAO/Comment.php

```diff
@@ -88,4 +109,27 @@ private static function validateMandatory($params) {
       );
     }
   }
+
+  /**
+   * Soft Deletes the comment with the given ID by setting the is_deleted column to 1
+   *
+   * @param int $id The ID of the comment to be soft deleted
+   *
+   * @return boolean
+   */
+  public static function softDelete($id) {
+    $comment = new self();
+    $comment->id = $id;
+
+    if ($comment->find()) {
+      while ($comment->fetch()) {
```

Comment:
Does is really need to a while? You're using the $id, so if we get more than one comment it means there's something really wrong happening :D 

To avoid the secondary call to `->fetch()` you can pass `true` to `find()` and it will call `fetch()` internally